### PR TITLE
feat(backoffice): [SFEQS-1764] Added ip addresses and fiscal codes functions

### PR DESCRIPTION
**Pull request by silvicir**
#### List of Changes

1. Added `POST` function in `/src/app/api/institutions/{institutionId}/api-keys/{apiKeyId}`
2. Added `listFiscalCodes` and `listIpAddresses` functions
3. Added unit tests

#### How Has This Been Tested?

Through unit tests

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-sign-backoffice-app/src/app/api/institutions/_lib/ip-address.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+const IpAddresses = z
+  .object({
+    cidrs: z
+      .string()
+      .regex(new RegExp(`(?:^${v4}$)|(?:^${v6}$)`))
+      .array()
+      .optional(),
+  })
+  .transform((res) => ({ cidrs: res.cidrs || [] }));
```
Comment (lucacavallaro) [inline apps/io-sign-backoffice-app/src/app/api/institutions/_lib/ip-address.ts:35]: Is this regex really necessary? could you not use directly `z.string().ip()` ?

ref: https://zod.dev/?id=ip-addresses

Comment (silvicir) [reply apps/io-sign-backoffice-app/src/app/api/institutions/_lib/ip-address.ts]: Not with the subnet mask

File: apps/io-sign-backoffice-app/src/app/api/institutions/_lib/ip-address.ts [hunk 2]
```diff
@@ -1,32 +1,23 @@
...
-(?::((?::${v6seg}){0,5}:${v4}|(?::${v6seg}){1,7}|:))           // ::2:3:4:5:6:7:8  ::2:3:4:5:6:7:8  ::8             ::1.2.3.4
-)(%[0-9a-zA-Z]{1,})?                                           // %eth0            %1
-\\/(12[0-8]|1[01][0-9]|[1-9]?[0-9])
-`
-  .replace(/\s*\/\/.*$/gm, "")
-  .replace(/\n/g, "")
-  .trim();
-
-export const ipAddressRegex = new RegExp(`(?:^${v4}$)|(?:^${v6}$)`);
+export const Cidr = z.custom<"cidr">((val) => {
```
Comment (lucacavallaro) [inline apps/io-sign-backoffice-app/src/app/api/institutions/_lib/ip-address.ts]:
```suggestion
export const CIDR = z.custom<"cidr">((val) => {
```

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: apps/io-sign-backoffice-app/src/app/api/institutions/_lib/fiscal-code.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { getCosmosConfig, getCosmosContainerClient } from "@/lib/cosmos";
+import { z } from "zod";
+
+export const fiscalCodeRegex = new RegExp(
+  "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"
+);
+
+const FiscalCodes = z
```
[the remark this answers is not in the document]
Comment (silvicir) [reply apps/io-sign-backoffice-app/src/app/api/institutions/_lib/fiscal-code.ts:8]: `FiscalCodesFromCosmosDb` ?

**Pull request merged**